### ISSUE TITLE
CUDNN_STATUS_EXECUTION_FAILED 

### ISSUE THREAD

**Issue by Vadim2S** [patrickltobing/cyclevae-vc-neuralvoco#2]
I am get strange CUDNN_STATUS_EXECUTION_FAILED error during VC model training.

OS: Docker container with Ubuntu 20.04
GPU: Two NVidia GeForce GTX Titan X
CUDA: 11.1
cuDNN installed, Nvidia tests passed OK.

No project installation (make) error.

Project: cycvae_mwdlp_vcc20

Data preparation passed OK.
Data preprocessing passed OK.

Set stage=4 (and comment others)
Set n_jobs=30 (and comment others)
Set GPU_device=0 (and comment others)
Set GPU_device_str=0 (and comment others)
Set n_gpus=1 (and comment others)

Run ./run.sh

After 1-2 minutes process aborted.

Log files attached.
[cvnm_log.zip](https://github.com/patrickltobing/cyclevae-vc-neuralvoco/files/6547084/cvnm_log.zip)

**Comment by patrickltobing** [comment]
Hi, I usually got `RuntimeError: cuDNN error: CUDNN_STATUS_EXECUTION_FAILED` if the GPU is ran out of memory.
Could you check your GPU status, e.g., with `nvidia-smi`, while running the script?

**Comment by Vadim2S** [comment]
Yes, of course. I am see something like 800, 3500, 5700, 1200 used GPU memory from 12000. Last iteration always has average  memory (~1200) and 100% GPU utilisation. Previuos iterations never has 100% GPU utilization. I am change args.batch-size to 7 from 30. No big changes.

**Comment by patrickltobing** [comment]
> I am see something like 800, 3500, 5700, 1200 used GPU memory from 12000.
OK, that looks normal.

> Last iteration always has average memory (~1200) and 100% GPU utilisation. Previuos iterations never has 100% GPU utilization.
I don't know what you meant by last iteration and previous iteration.
Do you mean that you had previously successfully run the script without error?

> I am change args.batch-size to 7 from 30. No big changes.
Could you try to change the batch_size to 1 and whether such error still comes up?

It might also be something related to the CUDA toolkit installation or so, I'm not sure. I have to reproduce the error.
Could you also let know your python version?

**Comment by Vadim2S** [comment]
> I don't know what you meant by last iteration and previous iteration.
Do you mean that you had previously successfully run the script without error?

No. No single success. I am place "print('Before crush')" line before crush line :-) and has several (~6 in stack) messages "Before crush" in log. I am presume here several succesfull cycle or multythread run some piece of code before crush. Last ones (and only last) always has 100% GPU utilization.

> Could you try to change the batch_size to 1 and whether such error still comes up?
> Could you also let know your python version?

Python version may be 3.8.5 (Ubuntu apt install latest). Sorry, today I am in the trip and can not do any changes or provide verified information before monday.

> It might also be something related to the CUDA toolkit installation

May be. But at least NVidia standard test "nMist..." work well and LD_LIBRARY_PATH + CUDA_HOME all right at "Before crush" point.

**Comment by patrickltobing** [comment]
Hi, I'm sorry for late response. Unfortunately, I could not reproduce your error.

Do you also found any problems on using GPU on your machine for other frameworks,
or is it the first time to have happened/encountered on your machine?

Anyway, I have tried to update the tested latest working commit to 1ef01de.
Please try to utilize the latest one, and do let me know if you still have any issues.
